cpu,stdlib: Fix Access Trace for Accessing Indices in SpatterGen

mahyarsamani opened the pull request:
This change fixes the way indices are generated in a multi generator setup.
It changes it from all cores generating the same trace of indices for accessing the index array to each core generating an interleaved subset of indices. 
For an example look below for traces (indices to index array) in a 2 core setup.

Before:
core_0: 0, 1, 2, 3, 4, 5, 6, 7, ...
core_1: 0, 1, 2, 3, 4, 5, 6, 7, ...
After:
core_0: 0, 1, 2, 3, 8, 9, 10, 11, ...
core_1: 4, 5, 6, 7, 12, 13, 14, 15, ...

Additionally, this change fixes the SpatterKernel class in the standard library to comply with the change in the SpatterGen source code.

mahyarsamani committed:
cpu, stdlib: Fixing access to index in SpatterGen.

This change fixes the way indices are generated in a multi
generator setup. It changes it from all cores generating the same
trace of indices for accessing the index array to each core
generating an interleaved subset of indices. For an example look
below for traces (indices to index array) in a 2 core setup.
Before:
core_0: 0, 1, 2, 3, 4, 5, 6, 7, ...
core_1: 0, 1, 2, 3, 4, 5, 6, 7, ...
After:
core_0: 0, 1, 2, 3, 8, 9, 10, 11, ...
core_1: 4, 5, 6, 7, 12, 13, 14, 15, ...

Additionally, this change fixes the SpatterKernel class in
the standard library to comply with the change in the
SpatterGen source code.

Change-Id: Idae69486eafafa21aaaad1239944c34ba8092087

powerjg reviewed (commented):
lgtm, but clang is unhappy about unused variables: https://github.com/gem5/gem5/actions/runs/9571285776/job/26388050223?pr=1258

BobbyRBruce committed:
cpu: Remove private fields from utility_structs.hh

These were causing clang compilation failures:

```sh
 [     CXX] src/cpu/pred/simple_btb.cc -> ALL/cpu/pred/simple_btb.o
In file included from src/cpu/testers/spatter_gen/spatter_gen.cc:29:
In file included from src/cpu/testers/spatter_gen/spatter_gen.hh:38:
src/cpu/testers/spatter_gen/utility_structs.hh:167:18: error: private field 'base' is not used [-Werror,-Wunused-private-field]
        uint32_t base;
                 ^
src/cpu/testers/spatter_gen/utility_structs.hh:211:14: error: private field 'index' is not used [-Werror,-Wunused-private-field]
    uint32_t index;
```

Change-Id: I927d277b6d1681593289f12447e735ad69ee44b8

BobbyRBruce commented:
I went ahead and removed the unused member variables.

BobbyRBruce committed:
cpu: Remove other member variable use

Change-Id: Idec02617f57c89db53af0706e763b4c43e51b1a2
